lib/surface/type_handler/list.ex

```diff
@@ -31,7 +31,7 @@ defmodule Surface.TypeHandler.List do
   defp handle_list_expr(name, expr) do
     quote generated: true do
       case unquote(expr) do
-        value when is_list(value) ->
+        value when is_list(value) or is_struct(value, Range) ->
           value
```

Comment:
@davydog187 maybe we should call `Enum.to_list/1` to make sure the value will be a list (unless this is already done somewhere else?). After all, the value is defined as `:list` so the user might have code depending on the value being a list, e.g. `List.first/1` and friends. This is similar to what we do when accepting keyword lists in props defined as `:map`, we accept it as syntactic sugar but always convert it to map. WDYT?

Reply:
Seems reasonable. Effectively you're suggesting doing an implicit cast so that when it crosses the component boundary, it's in the type expected. I buy that

Reply:
Exactly! If the user actually needs a range, then it's better to explicitly define the prop as `:range`.

Reply:
Ok, updated the code, feels like it should be tested though! Is `compiler_test.exs` the best place to test this?

Reply:
Done. Now the tests make more sense too. Good call @msaraiva